### PR TITLE
PPTP- 2761 - pass tax rate to frontend for content

### PR DESCRIPTION
**Pull request by mycp98**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [ ] `./precheck` was executed (Integration/Component/Unit tests)
 - [ ] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/services/TaxRateService.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   */
+  def lookupTaxRateForPeriod(periodEndDate: LocalDate): Double = {
+    if (periodEndDate.isBefore(appConfig.taxRegimeStartDate))
+      appConfig.taxRateBefore1stApril2022
+    else
+      appConfig.taxRateFrom1stApril2022
+  }
+
+
+  // thoughts? //todo not yes in use
```
Comment (mycp98) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/services/TaxRateService.scala:40]: we'll raise this in tech chat for feedback

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/services/WeightToPoundsConversionService.scala [hunk 2]
```diff
@@ -43,26 +42,15 @@ class WeightToPoundsConversionService @Inject()(appConfig: AppConfig) {
...
-    val taxRateForPeriod = lookupTaxRateForPeriod(periodEndDate)
+    val taxRateForPeriod = taxRateService.lookupTaxRateForPeriod(periodEndDate)
     val currency = BigDecimal(weightInKg) * taxRateForPeriod // tax rate is £ per kg
     currency.setScale(2, RoundingMode.DOWN)
   }
 
   // TODO allow for period / rate look up somehow
   def weightToCredit(weight: Long): BigDecimal = {
-    val currency = BigDecimal(weight) * appConfig.taxRateFrom1stApril2022
+    val currency = BigDecimal(weight) * taxRateService.lookupTaxRateForPeriod(LocalDate.now()) //todo: credits need to bring in a date
```
Comment (mycp98) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/services/WeightToPoundsConversionService.scala:52]: Tax rate change part B

**Review by mauro-fanton — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/config/AppConfig.scala [hunk 3]
```diff
@@ -71,8 +71,8 @@ class AppConfig @Inject() (config: Configuration, servicesConfig: ServicesConfig
   val desBearerToken: String = s"Bearer ${config.get[String]("microservice.services.des.bearerToken")}"
 
   val taxRegimeStartDate: LocalDate = LocalDate.of(2022, 4, 1)
-  val taxRateBefore1stApril2022: BigDecimal = BigDecimal(config.get[String]("taxRatePoundsPerKg"))
-  val taxRateFrom1stApril2022: BigDecimal = BigDecimal(config.get[String]("taxRatePoundsPerKg"))
+  val taxRateBefore1stApril2022: Double = config.get[Double]("taxRatePoundsPerKg")
```
Comment (lmcewanhmrc) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/config/AppConfig.scala:74]: I think this was correct before? ie money should be in BigDecimal, not double

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/services/PPTCalculationService.scala [hunk 4]
```diff
@@ -54,6 +55,7 @@ class PPTCalculationService @Inject()(
     val deductionsTotal: Long = totalExportedPlasticWeight + humanMedicinesPlasticWeight + recycledPlasticWeight
     val chargeableTotal: Long = scala.math.max(0, packagingTotal - deductionsTotal)
     val taxDue: BigDecimal = conversionService.weightToDebit(periodEndDate, chargeableTotal)
+    val taxRate: Double = taxRateService.lookupTaxRateForPeriod(periodEndDate)
```
Comment (lmcewanhmrc) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/services/PPTCalculationService.scala:59]: BigDecimal

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/config/AppConfig.scala (hunk 3 above)
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/config/AppConfig.scala:74]: @mauro-fanton here

**Review by liampan — APPROVED** [review]

**Pull request merged**